x-pack/plugins/alerting/server/task_runner/create_execution_handler.ts

```diff
@@ -159,40 +158,62 @@ export function createExecutionHandler<
-        id: action.id,
-        params: action.params,
-        spaceId,
-        apiKey: apiKey ?? null,
-        source: asSavedObjectExecutionSource({
-          id: alertId,
-          type: 'alert',
-        }),
-      });
+      const promise = new Promise(async (resolve, reject) => {
+        try {
+          await actionsClient.executeEphemeralTask({
+            taskType: `actions:${action.actionTypeId}`,
+            params: {
+              ...action.params,
+              taskParams: {
+                actionId: action.id,
+                apiKey,
+              },
+            },
+            state: {},
+          });
 
-      const namespace = spaceId === 'default' ? {} : { namespace: spaceId };
+          const namespace = spaceId === 'default' ? {} : { namespace: spaceId };
 
-      const event: IEvent = {
-        event: { action: EVENT_LOG_ACTIONS.executeAction },
-        kibana: {
-          alerting: {
-            instance_id: alertInstanceId,
-            action_group_id: actionGroup,
-            action_subgroup: actionSubgroup,
-          },
-          saved_objects: [
-            { rel: SAVED_OBJECT_REL_PRIMARY, type: 'alert', id: alertId, ...namespace },
-            { type: 'action', id: action.id, ...namespace },
-          ],
-        },
-      };
+          const event: IEvent = {
+            event: { action: EVENT_LOG_ACTIONS.executeAction },
+            kibana: {
+              alerting: {
+                instance_id: alertInstanceId,
+                action_group_id: actionGroup,
+                action_subgroup: actionSubgroup,
+              },
+              saved_objects: [
+                { rel: SAVED_OBJECT_REL_PRIMARY, type: 'alert', id: alertId, ...namespace },
+                { type: 'action', id: action.id, ...namespace },
+              ],
+            },
+          };
 
-      event.message = `alert: ${alertLabel} instanceId: '${alertInstanceId}' scheduled ${
-        actionSubgroup
-          ? `actionGroup(subgroup): '${actionGroup}(${actionSubgroup})'`
-          : `actionGroup: '${actionGroup}'`
-      } action: ${actionLabel}`;
-      eventLogger.logEvent(event);
+          event.message = `alert: ${alertLabel} instanceId: '${alertInstanceId}' scheduled ${
+            actionSubgroup
+              ? `actionGroup(subgroup): '${actionGroup}(${actionSubgroup})'`
+              : `actionGroup: '${actionGroup}'`
+          } action: ${actionLabel}`;
+          eventLogger.logEvent(event);
+          resolve(true);
+        } catch (err) {
+          return reject(err);
+        }
+      });
+      promises.push(promise);
+      // await actionsClient.executeEphemeralTask({
+      //   taskType: `actions:${action.actionTypeId}`,
+      //   params: {
+      //     ...action.params,
+      //     taskParams: {
+      //       actionId: action.id,
+      //       apiKey,
+      //     },
+      //   },
+      //   state: {},
+      // });
     }
+
+    await Promise.all(promises);
```

Comment:
It is my understanding of `Promise.all` that it will only resolve successfully if all promises resolve successfully and reject if a single promise rejects. According to [the docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all): `It rejects immediately upon any of the input promises rejecting or non-promises throwing an error, and will reject with this first rejection message / error.`.

Should we be adding logic inside the try/catch to still resolve on Error so that a single action execution failure won't cancel all other action executions?

Reply:
Alternatively, maybe we could use `Promise.allSettled`?

Reply:
@ymao1 Yea, great point. We should probably use the `asErr` pattern here to describe an error with an action to ensure all promises resolve properly